docs/source/en/modular_diffusers/getting_started.md

```diff
@@ -0,0 +1,1393 @@
+
+</Tip>
+
+## ModularPipeline
+
+`ModularPipeline` is the main interface to create and execute pipelines in the Modular Diffusers system.
+
+### Modular Repo
+
+`ModularPipeline` only works with modular repositories. You can find an example modular repo [here](https://huggingface.co/YiYiXu/modular-diffdiff).
+
+Instead of using a `model_index.json` to configure components loading in `DiffusionPipeline`. Modular repositories work with `modular_model_index.json`. Let's walk through the difference here.
+
+In standard `model_index.json`, each component entry is a `(library, class)` tuple:
+
+```py
+"text_encoder": [
+  "transformers",
+  "CLIPTextModel"
+],
+```
+
+In `modular_model_index.json`, each component entry contains 3 elements: `(library, class, loading_specs {})`
+
+- `library` and `class`: Information about the actual component loaded in the pipeline at the time of saving (will be `null` if not loaded)
+- `loading_specs`: A dictionary containing all information required to load this component, including `repo`, `revision`, `subfolder`, `variant`, and `type_hint`. 
+
+```py
+"text_encoder": [
+  null,  # library (same as model_index.json)
+  null,  # class (same as model_index.json)
+  {      # loading specs map (unique to modular_model_index.json)
+    "repo": "stabilityai/stable-diffusion-xl-base-1.0",  # can be a different repo
+    "revision": null,
+    "subfolder": "text_encoder",
+    "type_hint": [  # (library, class) for the expected component class
+      "transformers",  
+      "CLIPTextModel"
+    ],
+    "variant": null
+  }
+],
+```
+Some components may not have `repo` field, they cannot be loaded from a repository and can only be created with default config from the pipeline
+
+```py
+  "image_processor": [
+    "diffusers",
+    "VaeImageProcessor",
+    {
+      "type_hint": [
+        "diffusers",
+        "VaeImageProcessor"
+      ]
+    }
+  ],
+```
+Unlike standard repositories where components must be in subfolders within the same repo, modular repositories can fetch components from different repositories based on the `loading_specs` dictionary. e.g. the `text_encoder` component will be fetched from the "text_encoder" folder in `stabilityai/stable-diffusion-xl-base-1.0` while other components come from different repositories.
+
+
+### Creating a `ModularPipeline` from `ModularPipelineBlocks`
+
+Each `ModularPipelineBlocks` has an `init_pipeline` method that can initialize a `ModularPipeline` object based on its component and configuration specifications.
+
+Let's convert our `t2i_blocks` (which we created earlier) into a runnable `ModularPipeline`:
+
+```py
+# We already have this from earlier
+t2i_blocks = SequentialPipelineBlocks.from_blocks_dict(TEXT2IMAGE_BLOCKS)
+
+# Now convert it to a ModularPipeline
+modular_repo_id = "YiYiXu/modular-loader-t2i"
+t2i_pipeline = t2i_blocks.init_pipeline(modular_repo_id)
+```
+
```

Comment:
Can we spend a few sentences about how `t2i_blocks.init_pipeline(modular_repo_id)` is working under the hood as it's not obvious from the information available upto this point?